packages/modules/e3dc/device.py

```diff
@@ -0,0 +1,181 @@
+    component_config = E3dcBatSetup(configuration=E3dcBatConfiguration())
+    component_config.id = num
+    run_device_legacy(create_legacy_device_config(address1,
+                                                  address2,
+                                                  read_ext,
+                                                  pvmodul,
+                                                  num), component_config)
+
+
+def read_legacy_counter(address1: str,
+                        address2: str, read_ext: int,
+                        pvmodul: str,
+                        num: int):
+    component_config = E3dcCounterSetup(configuration=E3dcCounterConfiguration())
+    component_config.id = num
+    run_device_legacy(create_legacy_device_config(address1,
+                                                  address2,
+                                                  read_ext,
+                                                  pvmodul,
+                                                  num), component_config)
+
+
+def read_legacy_inverter(address1: str,
+                         address2: str, read_ext: int,
+                         pvmodul: str,
+                         num: int):
+    component_config = E3dcInverterSetup(configuration=E3dcInverterConfiguration())
+    component_config.id = num
+    run_device_legacy(create_legacy_device_config(address1,
+                                                  address2,
+                                                  read_ext,
+                                                  pvmodul,
+                                                  num), component_config)
+
+
+def read_legacy_batv19(address1: str,
+                       address2: str, read_ext: int,
+                       pvmodul: str,
+                       num: int) -> None:
+    # für openwbv19 können wmit der bisherigen parametrisierung zwei ip_addressen
+    # ausgelesen werden
+    # ebenso wird bei Speicheraufruf beides (Speicher und PV ausgelesen
+    # in openwb v2.0 geht nur noch eine IP adresse und die Pv muss
+    # separate ausgelesen werden
+    addresses = [address for address in [address1, address2] if address != "none"]
+    log.debug('e3dc IP-Adresse1: ' + address1)
+    log.debug('e3dc IP-Adresse2: ' + address2)
+    log.debug('e3dc read_ext: ' + str(read_ext))
+    log.debug('e3dc pvmodul: ' + pvmodul)
+    log.debug('e3dc id: ' + str(num))
+    soc = 0
+    count = 0
+    power = 0
+    pv_external = 0
+    pv = 0
+    pv_other = pvmodul != "none"
+    for address in addresses:
+        log.debug("Ip: %s, read_external %d pv_other %s", address, read_ext, pv_other)
+        count += 1
+        with modbus.ModbusTcpClient_(address, port=502) as client:
+            socwk, powerwk = read_bat(client)
+            soc += socwk
+            power += powerwk
+            pvwk, pv_externalwk = read_inverter(client, read_ext)
+            pv += pvwk
+            pv_external += pv_externalwk
+    soc = soc / count
```

Comment:
```suggestion
    soc /= len(addresses)
```
Spart dir die Variable `count` ein.